src/plonk/lookup.rs

```diff
@@ -242,28 +247,44 @@ impl<F: PrimeField> Arguments<F> {
     /// calculate the coefficients {m_i} in the log derivative formula
     /// m_i = sum_j \xi(w_j=t_i) assuming {t_i} have no duplicates
     fn evaluate_m(&self, l: &[F], t: &[F]) -> Vec<F> {
```

Comment:
can you explain a little bit about this algorithm and its complexity?

Reply:
# Algo

1. Creating a hash map (`l_elements`) to count occurrences of elements in `l`.
2. Iterating through `t`, and for each element `t_i`:
   - If `t_i` has already been processed (tracked using a hash set `processed_t`), it skips calculation
   - Otherwise, it looks up `t_i` in `l_elements` to find the count of occurrences, translating this count into an element of the field `F` (`F::from_u128`), and marks `t_i` as processed.

# Complexity

Let `n` be the length of `l` and `m` be the length of `t`. The complexity of this algorithm can
be broken down as follows:

- Building `l_elements`: O(n) as it involves a single iteration over `l` and insertions into a hash map.
- Iterating through `t` and looking up in `processed_t` and `l_elements`: O(m) assuming hash set and hash map operations are O(1) on average.
- Thus, the overall complexity is approximately O(n + m), which depends linearly on the sizes of the input slices `l` and `t`.

Reply:
Got it. The use of hashmap is much better than before

Reply:
With `O(n*m)` it's just hung for sha256 :sweat_smile: